### PR TITLE
book crud operations

### PR DESCRIPTION
**Pull request by ToStefan**
Book CRUD operations

### CODE REVIEW

File: src/main/java/hybrid/it/internship/library/service/IBookService.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package hybrid.it.internship.library.service;
+
+import hybrid.it.internship.library.entity.Book;
+
+import java.util.List;
+
+public interface IBookService {
```
Comment (babicmilan) [inline src/main/java/hybrid/it/internship/library/service/IBookService.java]: Dont need "I" in front. Think this is a c# convention.

File: src/main/java/hybrid/it/internship/library/service/IRentService.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package hybrid.it.internship.library.service;
+
+public interface IRentService {
```
Comment (babicmilan) [inline src/main/java/hybrid/it/internship/library/service/IRentService.java]: same here

File: src/main/java/hybrid/it/internship/library/service/impl/BookService.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import hybrid.it.internship.library.service.IBookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class BookService implements IBookService {
+
+    @Autowired
```
Comment (babicmilan) [inline src/main/java/hybrid/it/internship/library/service/impl/BookService.java]: Don't use @Autowired, use constructor injection instead(@AllArgsConstructor). Also fields should be final.

File: src/main/java/hybrid/it/internship/library/service/impl/RentService.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+import hybrid.it.internship.library.repository.RentRepository;
+import hybrid.it.internship.library.service.IRentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RentService implements IRentService {
+
+    @Autowired
```
Comment (babicmilan) [inline src/main/java/hybrid/it/internship/library/service/impl/RentService.java]: Same as in BookService

File: src/main/java/hybrid/it/internship/library/web/controller/BookController.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/api/book")
+public class BookController {
+
+    @Autowired
```
Comment (babicmilan) [inline src/main/java/hybrid/it/internship/library/web/controller/BookController.java]: same as in services

File: src/main/java/hybrid/it/internship/library/web/mapper/BookMapper.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class BookMapper implements Mapper<Book, BookDTO> {
+
+    @Override
+    public BookDTO toDTO(Book entity) {
+        BookDTO bookDTO = new BookDTO();
```
Comment (babicmilan) [inline src/main/java/hybrid/it/internship/library/web/mapper/BookMapper.java]: Try adding @Builder to the dto and using it in this mapper

**Review by babicmilan — CHANGES_REQUESTED** [review]

File: src/main/java/hybrid/it/internship/library/web/controller/BookController.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    private RentService rentService;
+
+    @Autowired
+    private BookMapper bookMapper;
+
+    @GetMapping
+    public ResponseEntity<List<BookDTO>> getBooks() {
+
+        List<BookDTO> bookDTOS = bookMapper.toDTO(bookService.getAll());
```
Comment (mspavkovic) [inline src/main/java/hybrid/it/internship/library/web/controller/BookController.java]: better to do conversion from entity to dto in Service layer (so not in controller)

**Review by mspavkovic — APPROVED** [review]

File: src/main/java/hybrid/it/internship/library/web/controller/BookController.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+        return new ResponseEntity<>(retVal, HttpStatus.OK);
+    }
+
+    @DeleteMapping(value = "/{id}")
+    public ResponseEntity<HttpStatus> deleteBook(@PathVariable("id") Long id) {
+
+        if (rentService.existsByBookId(id)) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        } else {
```
Comment (nemanja-blagojevic-valcon) [inline src/main/java/hybrid/it/internship/library/web/controller/BookController.java]: No need for else if return is above..

File: src/main/java/hybrid/it/internship/library/service/impl/BookServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    @Transactional
+    public BookDTO create(BookDTO bookDTO) {
+        return bookMapper.toDTO(bookRepository.save(bookMapper.toEntity(bookDTO)));
+    }
+
+    @Override
+    @Transactional
+    public BookDTO update(Long id, BookDTO bookDTO) {
+        final Book book = bookRepository.findById(id)
+                .orElseThrow(IllegalStateException::new);
```
Comment (nemanja-blagojevic-valcon) [inline src/main/java/hybrid/it/internship/library/service/impl/BookServiceImpl.java]: Use some more descriptive exception

**Review by nemanja-blagojevic-valcon — CHANGES_REQUESTED** [review]

**Pull request merged**